mobile/src/main/java/rs/readahead/washington/mobile/media/camera/LuminosityAnalyzer.kt

```diff
@@ -0,0 +1,39 @@
+/*Based upon com.robertlevonyan.demo.camerax.analyzer */
+package rs.readahead.washington.mobile.media.camera
+
+import android.util.Log
+import androidx.camera.core.ImageAnalysis
+import androidx.camera.core.ImageProxy
+import java.nio.ByteBuffer
+import java.util.concurrent.TimeUnit
+
+class LuminosityAnalyzer: ImageAnalysis.Analyzer {
+    private var lastAnalyzedTimestamp = 0L
+
+    private fun ByteBuffer.toByteArray(): ByteArray {
+        rewind()    // Rewind the buffer to zero
+        val data = ByteArray(remaining())
+        get(data)   // Copy the buffer into a byte array
+        return data // Return the byte array
+    }
+
+    override fun analyze(image: ImageProxy) {
+        val currentTimestamp = System.currentTimeMillis()
+        // Calculate the average luma no more often than every second
+        if (currentTimestamp - lastAnalyzedTimestamp >= TimeUnit.SECONDS.toMillis(1)) {
+            // Since format in ImageAnalysis is YUV, image.planes[0]
+            // contains the Y (luminance) plane
+            val buffer = image.planes[0].buffer
+            // Extract image data from callback object
+            val data = buffer.toByteArray()
+            // Convert the data into an array of pixel values
```

Comment:
Instead of using `remaining()` in `ByteArray(remaining())`, We can use `ByteArray(buffer.remaining())` to create the byte array with the exact size required.